apis/placement/v1alpha1/stagedupdate_types.go

```diff
@@ -0,0 +1,441 @@
+
+	// AfterStageTaskConditionApprovalRequestApproved indicates whether the approval request is approved.
+	// Its condition status can be one of the following:
+	// - "True" means the approval request is approved.
+	// - "False" means the approval request is not approved.
+	AfterStageTaskConditionApprovalRequestApproved AfterStageTaskConditionType = "ApprovalRequestApproved"
+
+	// AfterStageTaskConditionApprovalWaitTimeElapsed indicates whether the wait time after each stage is elapsed.
+	// We will fill the message of the condition of the remaining wait time if the status is "False".
+	// Its condition status can be one of the following:
+	// - "True" means the  wait time is elapsed.
+	// - "False" means the  wait time is not elapsed.
+	AfterStageTaskConditionApprovalWaitTimeElapsed AfterStageTaskConditionType = "WaitTimeElapsed"
+)
+
+// StagedUpdateRunConditionType identifies a specific condition of the StagedUpdateRun.
+// +enum
+type StagedUpdateRunConditionType string
+
+const (
+	// StagedUpdateRunConditionRunning indicates whether the staged update run is proceeding normally.
+	// Its condition status can be one of the following:
+	// - "True" means the staged update run is proceeding normally.
+	// - "False" means the staged update run is not proceeding normally.
+	// - "Unknown" means it is unknown.
+	StagedUpdateRunConditionRunning StagedUpdateRunConditionType = "Running"
+
+	// StagedUpdateRunConditionWaitingBetweenStage indicates whether the staged update run is waiting between stages.
+	// Its condition status can be one of the following:
+	// - "True" means the staged update run is waiting between stages.
+	// - "False" means the staged update run is not waiting between stages.
+	// - "Unknown" means it is unknown.
+	StagedUpdateRunConditionWaitingBetweenStage StagedUpdateRunConditionType = "WaitingBetweenStage"
+)
+
+// StagedUpdateRunList contains a list of StagedUpdateRun.
+// +kubebuilder:resource:scope="Namespaced"
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type StagedUpdateRunList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []StagedUpdateRun `json:"items"`
+}
+
+// +genclient
+// +genclient:nonNamespaced
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope="Namespaced",categories={fleet,fleet-placement}
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// StageApprovalRequest defines a request for the approval of the stage in the update.
+// Each snapshot MUST have the following labels:
+//   - `ParentStagedUpdateRun` which points to its owner stage update.
+//   - `TargetStage` which is the name of the stage that this approval request is for.
+//   - `IsLatestUpdateRunApproval` which indicates whether this approval request is the latest one related to this update run.
+type StageApprovalRequest struct {
```

Comment:
A little bit confused: how will this request get approved?

Reply:
user needs to patch the "Approved" condition of the request as true